CloudSQLReplicator/google/cloud/pontem/sql/replicator/util/cloudsql.py

```diff
@@ -0,0 +1,250 @@
+# Copyright 2018 The Pontem Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Wrapper for Cloud SQL API service proxy."""
+import uuid
+
+import google.auth
+
+from google.cloud.pontem.sql.replicator.util import gcp_api_util
+
+# Defaults for Cloud SQL instances
+DEFAULT_1ST_GEN_DB_VERSION = 'MYSQL_5_6'
+DEFAULT_2ND_GEN_DB_VERSION = 'MYSQL_5_7'
+DEFAULT_1ST_GEN_TIER = 'd2'
+DEFAULT_2ND_GEN_TIER = 'db-n1-standard-2'
+DEFAULT_1ST_GEN_REGION = 'us-central'
+DEFAULT_2ND_GEN_REGION = 'us-central1'
+
+# Cloud SQL Service
+SQL_ADMIN_SERVICE = 'sqladmin'
+SQL_ADMIN_SERVICE_VERSION = 'v1beta4'
+
+def build_sql_admin_service(credentials=None):
+    """Authorizes and sets custom CloudSQL Replicator user agent.
```

Comment:
Sorry to nit-pick, but I am still not sure how these two fit together:
"build_sql_admin_service"
"Authorizes and sets custom CloudSQL Replicator user agent."

In other words, your function name makes me believe that you are building a service but your comment makes me believe you're setting a user agent.

Reply:
Changed to "Builds Cloud SQL service proxy with custom user agent."